src/main/java/application/dao/DaoProgramer.java

```diff
@@ -0,0 +1,22 @@
+package application.dao;
+
+import application.entities.Programmer;
+
+import javax.sql.DataSource;
+import java.util.List;
+import java.util.Optional;
+
+interface DaoProgramer {
+
+    abstract void setdataSource(DataSource dataSource);
```

Comment:
modificator abstract is redundant here
read about methods in interface, they are abstract by default